src/realm/object-store/sync/app.cpp

```diff
@@ -564,17 +588,51 @@ std::vector<std::shared_ptr<SyncUser>> App::all_users() const
     return m_sync_manager->all_users();
 }
 
+std::string App::get_base_url() const
+{
+    std::lock_guard<std::mutex> lock(*m_route_mutex);
+    return m_base_url;
+}
+
+void App::update_base_url(std::optional<std::string> base_url, UniqueFunction<void(Optional<AppError>)>&& completion)
+{
+    std::string new_base_url = base_url.value_or(default_base_url);
+
+    log_debug("App::update_base_url: %1", new_base_url);
+
+    if (new_base_url.empty()) {
+        throw InvalidArgument("update_base_url: new base URL value cannot be empty");
+    }
+
+    // Validate the new base_url
+    if (!AppUtils::split_url(new_base_url)) {
+        throw sync::BadServerUrl(new_base_url);
+    }
+
+    {
+        std::lock_guard<std::mutex> lock(*m_route_mutex);
+        // Update the location if the base_url is different or a location update is already needed
+        m_location_updated = (new_base_url == m_base_url) && m_location_updated;
+    }
+    // If the new base_url is the same as the current base_url and the location has already been updated,
+    // then we're done
+    if (m_location_updated) {
```

Comment:
shouldn't this check be performed under the lock?

Reply:
Yes - added a temporary to store the value under lock for this check